### DISCUSSION TITLE
How to add font package in ezdxf

### DISCUSSION THREAD

**Discussion by luoyu-123** [mozman/ezdxf#340] [Q&A]
I want to add some new font packages to ezdxf, how can I add them
Best wishes

**Comment by mozman** [comment]
Hi!

I  don't understand what you mean to "add font package in/to ezdxf". Ezdxf does not provide any fonts, the drawing add-on relies on matplotlib and PyQt5 to render fonts. Please describe in more detail what you want to achieve.

**Reply by luoyu-123** [reply]
Dear Mr. Mozman, thank you very much for your reply. When I use the mtext entity of ezdxf, the style attribute in the dxfattribs parameter sets the font style, and in ezdxf there are also two files about font style, fonts.json and fonts.py, but in these two files There is no font style I want, how can I add the style I want to use to ezdxf.
E.g:
font styles like chinese_fs.tff

**Reply by mozman** [reply]
These are internal files and do not effect the content of DXF files or the result of what CAD applications display.

The font definitions for usage in DXF files are stored as `TextStyle` objects in `Drawing.styles`, see this tutorial how to create your own text styles: https://ezdxf.mozman.at/docs/tutorials/text.html#new-text-style

**Reply by luoyu-123** [reply]
thank you
